hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -367,6 +368,7 @@ object HoodieSparkSqlWriter {
                 hoodieRecords
               }
             client.startCommitWithTime(instantTime, commitActionType)
+            SparkSampleWritesUtils.overwriteRecordSizeEstimateIfNeeded(jsc, hoodieRecords, client.getConfig, instantTime)
```

Comment:
should we call this in UpsertPartitioner where we try to determine the avg record size? so that all writers can leverage. as of now, we explicitly call from both spark sql writer and deltastreamer.

Reply:
synced up offline. we can keep it at higher layer.